src/main/java/org/mockito/junit/MockitoRule.java

```diff
@@ -76,4 +78,24 @@
      * @since 2.1.0
      */
     MockitoRule silent();
+
+    /**
+     * Strict stubbing is a new feature introduced in Mockito 2.3.
+     * Rule will increase productivity and debuggability by strictly validating stubs. It adds following behavior:
+     * <ol>
+     *     <li>Test fails early when code under test invokes stubbed method with different arguments (see {@link PotentialStubbingProblem}).</li>
+     *     <li>Test fails when there are any unused stubs declared (see {@link UnnecessaryStubbingException}).</li>
+     *     <li>If you use {@link org.mockito.Mockito#verifyNoMoreInteractions(Object...)} ;}
+     *     you no longer need to explicitly verify stubbed invocations. They are automatically verified.
+     *     This increases DRY-ness ("don't repeat yourself") of the tests making them cleaner and easier to maintain.</li>
+     * </ol>
+     * We are very eager to hear feedback about "strict stubbing" feature, let us know by commenting on GitHub
+     * <a href="https://github.com/mockito/mockito/issues/769">issue 769</a>.
+     * Strict stubbing is an attempt to improve testability and productivity with Mocktio. Tell us what you think!
+     * <p>
+     * This feature is tentatively planned to be turned on by default in Mockito v3.
+     *
+     * @since 2.3.0
+     */
+    MockitoRule strictStubs();
```

Comment:
It came to me earlier, but I have doubts on the word _strict_. When I first opened this PR, I thought this was about easymock style stubs. Then I realised it was not.

_Strict stubs_ may refer / apply to different things regarding stubs, I think a different name should be used. I'm halfway re-reviewing it (it's long) and my understanding of this change is that it is much more about potential stubbing problem detection than about strict mocks.

Also I'd add `@Incubating` there. 

Finally I'm thinking about rule builders, and I'd like to avoid a proliferation of this kind of API right now. Typically what I have in mind is rule builder where I can

* alter the mockito configuration (i.e. plugins override)
* alter the behaviour of mock (e.g. the easymock style)
* alter the logger
* etc.